contracts/vesting-lp/src/tests/integration.rs

```diff
@@ -0,0 +1,1248 @@
+            },
+            VestingAccount {
+                address: user2.to_string(),
+                schedules: vec![VestingSchedule {
+                    start_point: VestingSchedulePoint {
+                        time: app.block_info().time.seconds(),
+                        amount: Uint128::zero(),
+                    },
+                    end_point: Some(VestingSchedulePoint {
+                        time: app
+                            .block_info()
+                            .time
+                            .plus_seconds(100 * BLOCK_TIME)
+                            .seconds(),
+                        amount: Uint128::new(1000),
+                    }),
+                }],
+            },
+        ],
+    };
+
+    app.execute_contract(
+        owner,
+        vesting_instance.clone(),
+        &native_msg,
+        &coins(1200, VESTING_TOKEN),
+    )
+    .unwrap();
+
+    let query = QueryMsg::AvailableAmount {
+        address: user1.to_string(),
+    };
+
+    for _ in 1..=10 {
+        let vesting_res: Uint128 = app
+            .wrap()
+            .query_wasm_smart(vesting_instance.clone(), &query)
+            .unwrap();
+        assert_eq!(vesting_res, Uint128::new(0u128));
+
+        app.update_block(|b| {
+            b.height += 10;
+            b.time = b.time.plus_seconds(10 * BLOCK_TIME)
+        });
+
+        let vesting_res: Uint128 = app
+            .wrap()
+            .query_wasm_smart(vesting_instance.clone(), &query)
+            .unwrap();
+        assert_eq!(vesting_res, Uint128::new(20u128));
+
+        let msg = ExecuteMsg::Claim {
+            recipient: None,
+            amount: None,
+        };
+        let _res = app
+            .execute_contract(user1.clone(), vesting_instance.clone(), &msg, &[])
+            .unwrap();
+
+        let vesting_res: Uint128 = app
+            .wrap()
+            .query_wasm_smart(vesting_instance.clone(), &query)
+            .unwrap();
+        assert_eq!(vesting_res, Uint128::new(0u128));
+    }
+    app.update_block(|b| {
+        b.height += 100;
+        b.time = b.time.plus_seconds(100 * BLOCK_TIME)
+    });
+    let vesting_res: Uint128 = app
+        .wrap()
+        .query_wasm_smart(vesting_instance.clone(), &query)
+        .unwrap();
+    assert_eq!(vesting_res, Uint128::new(0u128));
+
+    let query_user_unclamed = UnclaimedAmountAtHeight {
+        address: user1.to_string(),
+        height: start_block_height - 1,
+    };
+    let vesting_res: Uint128 = app
+        .wrap()
+        .query_wasm_smart(vesting_instance.clone(), &query_user_unclamed)
+        .unwrap();
+    assert_eq!(vesting_res, Uint128::new(0u128));
+
+    let query_total_unclamed = UnclaimedTotalAmountAtHeight {
+        height: start_block_height - 1,
+    };
+    let vesting_res: Uint128 = app
+        .wrap()
+        .query_wasm_smart(vesting_instance.clone(), &query_total_unclamed)
+        .unwrap();
+    assert_eq!(vesting_res, Uint128::new(0u128));
```

Comment:
we check the amount at certain height, and as you can see above, the height for the check - `start_block_height - 1`, so `0` the exactly result we are expecting at the height, because we created a vestingSchedule after the height